src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -258,6 +263,12 @@ private async Task ProjectConfigurationFilePathStore_ChangedAsync(ProjectConfigu
 
     public Task OnLoadedAsync()
     {
+        // If the user hasn't turned on the Cohost server, then don't disable the Razor server
+        if (_languageServerFeatureOptions.DisableRazorLanguageServer && _languageServerFeatureOptions.UseRazorCohostServer)
```

Comment:
>_languageServerFeatureOptions.DisableRazorLanguageServer && _languageServerFeatureOptions.UseRazorCohostServer [](http://example.com/codeflow?start=12&length=110)

Nit: Out of curiosity - does it ever make sense to have both of them running? Could we just use the UseRazorCohostServer flag to turn off the "normal" Razor language server?

Reply:
Having both means no loss of user functionality, whereas right now if we disable the normal server then only document color and semantic tokens work, and even then not all of semantic tokens work because we never tell the editor to refresh etc. Sometimes thats useful, like if someone is working on making semantic tokens refresh etc., but sometimes its annoying. Whether or not this is a useful distinction will entirely depend on how long all of the cohost pieces take to land, in what order, and what we want to do about rolling it out. Adding the feature flag was easier than trying to plan the future :)

Reply:
Fair enough